scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/synchronization/AOSynchronization.java

```diff
@@ -0,0 +1,824 @@
+                return false;
+            }
+        });
+        for (File file : dbChannelFiles) {
+            try {
+                logger.info("Deleting " + file);
+                file.delete();
+            } catch (Exception e) {
+                logger.info("Error while deleting file during database cleanup", e);
+            }
+        }
+        return true;
+    }
+
+    public boolean close() {
+        inMemoryChannels.clear();
+        if (recMan != null) {
+            try {
+                logger.info("Closing Record Manager");
+                recMan.close();
+                recMan = null;
+            } catch (IOException e) {
+
+            }
+        }
+        return true;
+    }
+
+    private <T extends Serializable> Channel<T> getChannel(String name) throws InvalidChannelException {
+        if (inMemoryChannels.containsKey(name)) {
+            return inMemoryChannels.get(name);
+        } else if (persistedChannels.containsKey(name)) {
+            return persistedChannels.get(name);
+        }
+        throw new InvalidChannelException("Channel " + name + " does not exist");
+    }
+
+    @Override
+    public boolean createChannel(String originator, TaskId taskid, String name, boolean isPersistent)
+            throws IOException {
+        try {
+            boolean alreadyExistingChannel = deleteChannel(originator, taskid, name);
+            Channel newChannel = new Channel();
+            if (isPersistent) {
+                persistedChannels.put(name, newChannel);
+                logWithContext(taskid, null, "Created new persistent channel \'" + name + "\'", Level.INFO);
+                recMan.commit();
+            } else {
+                logWithContext(taskid, null, "Created new memory channel \'" + name + "\'", Level.INFO);
+                inMemoryChannels.put(name, newChannel);
+            }
+            return !alreadyExistingChannel;
+        } catch (InvalidChannelException e) {
+            throw new IllegalStateException(e);
+        }
+    }
+
+    @Override
+    public boolean deleteChannel(String originator, TaskId taskid, String name) throws IOException {
+        try {
+            if (persistedChannels.containsKey(name)) {
+                persistedChannels.remove(name);
+                logWithContext(taskid, null, "Deleted persistent channel \'" + name + "\'", Level.INFO);
+                recMan.commit();
+                return true;
+            } else if (inMemoryChannels.containsKey(name)) {
+                inMemoryChannels.remove(name);
+                logWithContext(taskid, null, "Deleted memory channel \'" + name + "\'", Level.INFO);
+                return true;
+            } else {
+                return false;
+            }
+        } catch (InvalidChannelException e) {
+            throw new IllegalStateException(e);
+        }
+    }
+
+    @Override
+    public <T extends Serializable> boolean createChannelIfAbsent(String originator, TaskId taskid, String name,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') T is not used in the method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2326)